src/executor/query/AggregateExecutor.cpp

```diff
@@ -27,6 +27,32 @@ folly::Future<Status> AggregateExecutor::execute() {
     QueryExpressionContext ctx(ectx_);
 
     std::unordered_map<List, std::vector<std::unique_ptr<AggData>>, std::hash<nebula::List>> result;
+
+    // generate default result when input dataset is empty
+    if (UNLIKELY(!iter->valid())) {
```

Comment:
I think it's extra code. If your agg function don't apply at all it will result the default value of course.

Reply:
Be consistent with cypher:
```
neo4j@neo4j> match (v:player) where v.age<(-3) return sum(v.age);
+------------+
| sum(v.age) |
+------------+
| 0          |
+------------+

```

Reply:
I means your agg function should result default value if don't apply at all.

Reply:
If the aggregate function is not applied, then additional interfaces or data structures are required to store the default values ​​of various functions, which is contradictory to the design of `AggregateManager`. 

If you apply the function, you need to reallocate memory of `AggData`. bcz, `AggData` is dynamically allocated and managed with iteration in the for block.